go/vrf/conversion.go

```diff
@@ -79,34 +78,30 @@ func secg1EncodeCompressed(curve elliptic.Curve, x, y *big.Int) []byte {
 
 var errInvalidPoint = errors.New("invalid point")
 
-func secg1Decode(curve elliptic.Curve, data []byte) (x, y *big.Int, err error) {
+func unmarshalCompressed(curve elliptic.Curve, data []byte) (x, y *big.Int, err error) {
 	byteLen := (curve.Params().BitSize + 7) >> 3
-	if (data[0] &^ 1) != 2 {
+	if (data[0] &^ 1) != 2 { // compressed form
 		return nil, nil, errors.New("unrecognized point encoding")
 	}
 	if len(data) != 1+byteLen {
 		return nil, nil, errors.New("invalid length for curve")
 	}
 
 	// Based on Routine 2.2.4 in NIST Mathematical routines paper
-	params := curve.Params()
-	tx := new(big.Int).SetBytes(data[1 : 1+byteLen])
-	y2 := y2(params, tx)
-	sqrt := defaultSqrt
-	ty := sqrt(y2, params.P)
-	if ty == nil {
+	x = new(big.Int).SetBytes(data[1:])
+	y2 := y2(curve.Params(), x)
```

Comment:
This is a pre-existing problem, but I would rename the function `y2` to something more descriptive. This line "y2 := y2(...)" is confusing to me. Also, it looks like the only place the y2 function is called is to calculate y**2 from x. What you really care about is getting y from x, so the square root could be moved into the calculateY function.

Reply:
I've renamed it "polynomial" to match https://github.com/golang/go/pull/35110
Is that clearer?